pkg/webhook/server.go

```diff
@@ -47,27 +48,19 @@ type ServerOptions struct {
 	// the user is responsible to mount the secret to the this location for the server to consume.
 	CertDir string
 
-	// err will be non-nil if there is an error occur during initialization.
-	err error // nolint: structcheck
-}
-
-// Server is an admission webhook server that can serve traffic and
-// generates related k8s resources for deploying.
-type Server struct {
-	// Name is the name of server
-	Name string
+	// TODO(directxman12): should we make the mux configurable?
```

Comment:
Is there a use case in your head that the user may want to configure mux?

Reply:
if you want to serve on an existing handler more easily.  It's not clear that there's a concrete usecase, but I can imagine someone saying something at some points.

Reply:
(e.g. your comment below about `Handle`)

Reply:
Ha, I got what you were saying from https://github.com/kubernetes-sigs/controller-runtime/pull/323/commits/38e611d7f34a259b36892bbdad7f64800123ca9d#r258328244.
Then IMO we should make `webhookMux` field public.

Reply:
> (e.g. your comment below about `Handle`)

We probably should make `webhookMux` field configurable.